### PR TITLE
benchmarks on dataproc

### PR DESCRIPTION
**Pull request by leewyang**

### CODE REVIEW

**Comment by leewyang** [comment]
skipping build since this PR doesn't impact it.

**Review by eordentlich — COMMENTED** [review]
Looks good.   Some minor comments.
One question for cpu side.  Do you see any warnings about BLAS and LAPACK libraries not being set up?

File: python/benchmark/dataproc/README.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# Benchmarking on Dataproc
+
+This directory contains shell scripts for running larger scale benchmarks on a Google Dataproc cluster.  You will need a Dataproc account to run them.  The benchmarks use datasets synthetically generated using [gen_data.py](../gen_data.py).  For convenience, these have been precomputed and currently stored in the public S3 bucket `spark-rapids-ml-bm-datasets-public`.  The benchmark scripts are currently configured to read the data from there.
```
[the remark this answers is not in the document]
Comment (leewyang) [reply python/benchmark/dataproc/README.md]: Good catch, this was left over from original attempt to read from S3... fixed now.

File: python/benchmark/dataproc/start_cluster.sh [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    echo "Please export GCS_BUCKET per README.md"
+    exit 1
+fi
+
+BENCHMARK_HOME=${BENCHMARK_HOME:-${GCS_BUCKET}/benchmark}
+CUDA_VERSION=${CUDA_VERSION:-11.8}
+RAPIDS_VERSION=${RAPIDS_VERSION:-23.02}
+
+gpu_args=$(cat <<EOF
+--master-accelerator type=nvidia-tesla-t4,count=1
```
[the remark this answers is not in the document]
Comment (leewyang) [reply python/benchmark/dataproc/start_cluster.sh:20]: Presumaby, yes, although @wbo4958 has a [PR](https://github.com/NVIDIA/spark-rapids-ml/pull/233) to fix this now.

**Comment by leewyang** [comment]
> One question for cpu side. Do you see any warnings about BLAS and LAPACK libraries not being set up?

Yes, I see the following in the logs
```
# egrep "BLAS|LAPACK" *
kmeans_1.out:23/04/14 20:23:50 WARN com.github.fommil.netlib.BLAS: Failed to load implementation from: com.github.fommil.netlib.NativeSystemBLAS
kmeans_1.out:23/04/14 20:23:50 WARN com.github.fommil.netlib.BLAS: Failed to load implementation from: com.github.fommil.netlib.NativeRefBLAS
linear_regression_elasticnet_1.out:23/04/14 22:41:12 WARN com.github.fommil.netlib.BLAS: Failed to load implementation from: com.github.fommil.netlib.NativeSystemBLAS
linear_regression_elasticnet_1.out:23/04/14 22:41:12 WARN com.github.fommil.netlib.BLAS: Failed to load implementation from: com.github.fommil.netlib.NativeRefBLAS
linear_regression_noreg_1.out:23/04/14 22:31:47 WARN com.github.fommil.netlib.BLAS: Failed to load implementation from: com.github.fommil.netlib.NativeSystemBLAS
linear_regression_noreg_1.out:23/04/14 22:31:48 WARN com.github.fommil.netlib.BLAS: Failed to load implementation from: com.github.fommil.netlib.NativeRefBLAS
linear_regression_noreg_1.out:23/04/14 22:31:48 WARN com.github.fommil.netlib.LAPACK: Failed to load implementation from: com.github.fommil.netlib.NativeSystemLAPACK
linear_regression_noreg_1.out:23/04/14 22:31:48 WARN com.github.fommil.netlib.LAPACK: Failed to load implementation from: com.github.fommil.netlib.NativeRefLAPACK
linear_regression_ridge_1.out:23/04/14 22:50:20 WARN com.github.fommil.netlib.BLAS: Failed to load implementation from: com.github.fommil.netlib.NativeSystemBLAS
linear_regression_ridge_1.out:23/04/14 22:50:20 WARN com.github.fommil.netlib.BLAS: Failed to load implementation from: com.github.fommil.netlib.NativeRefBLAS
linear_regression_ridge_1.out:23/04/14 22:50:21 WARN com.github.fommil.netlib.LAPACK: Failed to load implementation from: com.github.fommil.netlib.NativeSystemLAPACK
linear_regression_ridge_1.out:23/04/14 22:50:21 WARN com.github.fommil.netlib.LAPACK: Failed to load implementation from: com.github.fommil.netlib.NativeRefLAPACK
pca_1.out:23/04/14 22:19:30 WARN com.github.fommil.netlib.LAPACK: Failed to load implementation from: com.github.fommil.netlib.NativeSystemLAPACK
pca_1.out:23/04/14 22:19:30 WARN com.github.fommil.netlib.LAPACK: Failed to load implementation from: com.github.fommil.netlib.NativeRefLAPACK
```

**Comment by leewyang** [comment]
Resolved the issue with the init scripts, so should be good to go now.

File: python/benchmark/dataproc/README.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
+# Benchmarking on Dataproc
+
+This directory contains shell scripts for running larger scale benchmarks on a Google Dataproc cluster.  You will need a Dataproc account to run them.  The benchmarks use datasets synthetically generated using [gen_data.py](../gen_data.py).  For convenience, these have been precomputed and currently stored in the public GCS bucket `gs://spark-rapids-ml-benchmarking/datasets`.  The benchmark scripts are currently configured to read the data from there.
```
Comment (eordentlich) [inline python/benchmark/dataproc/README.md:3]: The gs bucket is not public.    Let's not make it public for now.   Somehow reword this comment.

Comment (leewyang) [reply python/benchmark/dataproc/README.md]: Updated.

**Pull request merged**